### PR TITLE
Changes up to validator extensions.

### PR DESCRIPTION
**Pull request by powdercloud**
Release Notes:
- amp-jwplayer support (github #2734)
- Require amp-analytics.js for amp-user-notification and amp-access (github #2701)
- rel= attribute now a blacklist, so values not covered by the blacklist are allowed by default
- Validator supports amp-sidebar.
- HTML Parser now parses <? etc. as tags and the Validator rejects them (e.g. ?php).

### CODE REVIEW

**Comment by powdercloud** [comment]
@Gregable @honeybadgerdontcare

**Comment by powdercloud** [comment]
@dvoytenko

**Comment by powdercloud** [comment]
Sorry I found a small bug in build.py (I fat-fingered the glob). Will fix this later and add another commit here, and say when it's ready.

**Comment by powdercloud** [comment]
Should be ready for CLA confirmation / review / etc. Thanks for the patience.

**Comment by honeybadgerdontcare** [comment]
@googlebot confirmed

**Comment by Gregable** [comment]
@googlebot confirmed

**Comment by Gregable** [comment]
Can you go through and provide a quick summary any changes affecting PASS/FAIL in the PR description? Example https://github.com/ampproject/amphtml/pull/2595 . This'll be used by @erwinmombay for his release notes.

File: validator/testdata/feature_tests/amp_sidebar.out [hunk 1]
```diff
@@ -0,0 +1,4 @@
+FAIL
```
Comment (dvoytenko) [inline validator/testdata/feature_tests/amp_sidebar.out]: Aren't these supposed to go into `extensions/amp-sidebar/0.1/test` and such?

**Comment by dvoytenko** [comment]
@powdercloud Not sure I'm seeing changes that are all relevant to extensions. I see bunch of parse_css changes and such. Is this intended?

File: validator/testdata/feature_tests/amp_sidebar.out (hunk 1 above)
Comment (powdercloud) [inline validator/testdata/feature_tests/amp_sidebar.out]: Yes, but not in this change. I've started by just splitting validator.protoascii for now and will get the tests done seperately.

**Comment by powdercloud** [comment]
@dvoytenko Yes, those are all the recent changes that we're putting out, not just the extensions stuff. I'll need to update the description.

**Comment by powdercloud** [comment]
Sigh, I'll start a new one, please ignore the rest here.

**Comment by powdercloud** [comment]
OK, I think it's all right after all.
@Gregable @honeybadgerdontcare Still need confirmation for the CLA.

**Comment by Gregable** [comment]
@googlebot confirmed again

**Comment by dvoytenko** [comment]
LGTM

**Comment by honeybadgerdontcare** [comment]
@googlebot confirmed

**Comment by powdercloud** [comment]
@cramforce Thank you!!

**Pull request merged**